p2p/src/net/mock/transport/traits.rs

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) 2022 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::{fmt::Debug, hash::Hash};
+
+use async_trait::async_trait;
+
+use crate::{net::mock::types::Message, Result};
+
+/// An abstraction layer for creating and opening connections.
+#[async_trait]
+pub trait MockTransport: Send + 'static {
+    /// An address type.
+    type Address: Copy + Clone + Debug + Eq + Hash + Send + Sync + ToString;
```

Comment:
This is just for testing so it's not that necessary... but why does Address have to implement the Copy trait? Seems like a huge limitation.

Reply:
I don't like this either and it isn't consistent with the `NetworkingService::Address` requirements, but the previous mocks implementation relies on copy-able addresses, so I decided to preserve this requirement in order to introduce fewer changes (such as `address.clone()`).

I'm going to remove the `Copy` bound.